File: spring-cloud-starter/src/main/java/com/jmsoftware/maf/springcloudstarter/configuration/MinioConfiguration.java
package com.jmsoftware.maf.springcloudstarter.configuration;

import com.jmsoftware.maf.springcloudstarter.helper.MinioHelper;
import io.minio.MinioClient;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;

/**
 * <h1>MinioConfiguration</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com, date: 6/7/21 9:58 PM
 **/
@Slf4j
@EnableConfigurationProperties({
        MinioProperties.class
})
@ConditionalOnProperty({"minio.enabled"})
public class MinioConfiguration {
    @Bean
    public MinioClient minioClient(MinioProperties minioProperties) {
        log.warn("Initial bean: '{}'", MinioClient.class.getSimpleName());
        return MinioClient.builder()
                .endpoint(minioProperties.getEndpoint(), minioProperties.getPort(),
                          minioProperties.getSecure())
                .credentials(minioProperties.getAccessKey(), minioProperties.getSecretKey())
                .build();
    }

    @Bean
    public MinioHelper minioHelper(MinioClient minioClient) {
        log.warn("Initial bean: '{}'", MinioHelper.class.getSimpleName());
        return new MinioHelper(minioClient);
    }
}

File: spring-cloud-starter/src/main/kotlin/com/jmsoftware/maf/springcloudstarter/minio/MinioConfiguration.kt
package com.jmsoftware.maf.springcloudstarter.minio

import com.jmsoftware.maf.common.util.logger
import io.minio.MinioClient
import org.springframework.boot.autoconfigure.condition.ConditionalOnClass
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean

/**
 * # MinioConfiguration
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/14/22 3:25 PM
 */
@ConditionalOnProperty("minio.enabled")
@ConditionalOnClass(MinioClient::class)
@EnableConfigurationProperties(MinioProperties::class)
class MinioConfiguration {
    companion object {
        private val log = logger()
    }

    @Bean
    fun minioClient(minioProperties: MinioProperties): MinioClient {
        log.warn("Initial bean: `${MinioClient::class.java.simpleName}`" )
        return MinioClient.builder()
            .endpoint(
                minioProperties.endpoint, minioProperties.port,
                minioProperties.secure
            )
            .credentials(minioProperties.accessKey, minioProperties.secretKey)
            .build()
    }

    @Bean
    fun minioHelper(minioClient: MinioClient): MinioHelper {
        log.warn("Initial bean: `${MinioHelper::class.java.simpleName}`" )
        return MinioHelper(minioClient)
    }
}
